tests/syntax-tests/test_custom_assets.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+set -o errexit -o nounset -o pipefail
+
+### HELPER VARS
+
+custom_syntaxes_dir="$(bat --config-dir)/syntaxes"
+
+installed_custom_syntax="${custom_syntaxes_dir}/BatTestCustomAssets.sublime-syntax"
+
+custom_syntax_args=(
+    "--language=BatTestCustomAssets"
+    "tests/syntax-tests/source/BatTestCustomAssets/NoColorsUnlessCustomAssetsAreUsed.battestcustomassets"
+)
+
+integrated_syntax_args=(
+    "--language=Rust"
+    "examples/simple.rs"
+)
+
+### HELPER FUNCTIONS
+
+echo_step() {
+    echo -e "\n== $1 =="
+}
+
+fail_test() {
+    echo "FAIL: $1"
+    exit 1
+}
+
+### TEST STEPS
+
+echo_step "TEST: Make sure 'BatTestCustomAssets' is not part of integrated syntaxes"
+bat -f "${custom_syntax_args[@]}" &&
```

Comment:
If you are unfamiliar with the `"${custom_syntax_args[@]}"` syntax, you are in good company. But it is what is recommended by https://github.com/koalaman/shellcheck/wiki/SC2086#exceptions